src/AppInstallerCommonCore/Deployment.cpp

```diff
@@ -94,6 +107,92 @@ namespace AppInstaller::Deployment
         WaitForDeployment(deployOperation, id, callback);
     }
 
+    bool AddPackageWithDeferredFallback(
+        const std::string& uri,
+        bool skipSmartScreen,
+        IProgressCallback& callback)
+    {
+        PackageManager packageManager;
+
+        // In the event of a failure we want to ensure that the package is not left on the system.
+        Msix::MsixInfo packageInfo{ uri };
+        std::wstring packageFullName = packageInfo.GetPackageFullNameWide();
+        auto removePackage = wil::scope_exit([&]() {
+            try
+            {
+                RemovePackage(Utility::ConvertToUTF8(packageFullName), callback);
+            }
+            CATCH_LOG();
+        });
+
+        Uri uriObject(Utility::ConvertToUTF16(uri));
+
+        if (!skipSmartScreen)
+        {
+            // The only way to get SmartScreen is to use RequestAddPackageAsync, so we will have to start with that.
+            size_t id = GetDeploymentOperationId();
+            AICLI_LOG(Core, Info, << "Starting RequestAddPackageAsync operation #" << id << ": " << uri);
+
+            DeploymentOptions options = DeploymentOptions::None;
+            // Optimization to keep files if the package is in use. Only available in a newer OS per:
+            // https://docs.microsoft.com/en-us/uwp/api/Windows.Management.Deployment.DeploymentOptions
+            if (Runtime::IsCurrentOSVersionGreaterThanOrEqual(Utility::Version{ "10.0.18362.0" }))

```

Comment:
should we keep consts with all versions we reference and applicable names (i.e. win11 or 20H1 etc.)?

Reply:
Possibly.  I was intentionally tying it to the documented version in which this particular flag was made available rather than the OS release name/timeframe.  If we need this kind of check more in the future I think it would be fine to implement then.